### PR TITLE
BE query records with sql

### PR DESCRIPTION
**Pull request by AdrieneTZ**
- Rename routes and controller files name
- Generate seed records with a random timestamp during the last two years from now
- Add function: get the current user's all saving records
- Query all income, expense and saving records with SQL

### CODE REVIEW

File: apps/back-end/controllers/record-controller.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+      // uncommend after adding authentication process
+      // const userId = req.user.id
+      // remove above after adding authentication process
+      const userIds = await prisma.$queryRaw`SELECT id FROM "User";`
+      const userId = userIds.map(({ id }) => id)[0]
+
+      // Get categoryIds
+      const result =
+        await prisma.$queryRaw`SELECT id FROM "Category" WHERE NOT "mainCategory" = 'Income' AND NOT "mainCategory" = 'Savings';`
```
Comment (davipon) [inline apps/back-end/controllers/record-controller.js:48]: The exclusive query (`WHERE **NOT** "mainCategory"`) might work in the current sample with fewer categories.
It'd be better to use explicit syntax to query expense records.

File: apps/back-end/controllers/record-controller.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          'missing title or amount or category id to create a new expense record'
+        )
+
+      await prisma.record.create({
+        data: {
+          title,
+          amount,
+          note,
+          categoryId,
+          userId: 13,
```
Comment (davipon) [inline apps/back-end/controllers/record-controller.js:25]: We'd better work on the user system in the next PR so we can use the right `userId`.

File: apps/back-end/controllers/record-controller.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      const theRecord = await prisma.record.findUnique({
+        where: {
+          id: recordId,
+        },
+      })
+      theRecord ? theRecord : res.json('the record does not exist')
+
+      await prisma.record.update({
+        where: { id: recordId },
+        data: { title, amount, note, categoryId, userId: 13 },
```
Comment (davipon) [inline apps/back-end/controllers/record-controller.js:96]: We'd better work on the user system in the next PR to use the right `userId`.

**Review by davipon — APPROVED** [review]
Should we seed data again before starting the server?

File: apps/back-end/controllers/record-controller.js (hunk 1 above)
Comment (AdrieneTZ) [reply apps/back-end/controllers/record-controller.js:48]: I'm thinking about separating our one category model into three models, income, expense and saving model. In the category model, I named the main category `Income`, but other expense categorise did not belong to any main category called `expense`. This design would make some difficulties while querying data.

File: apps/back-end/controllers/record-controller.js (hunk 2 above)
Comment (AdrieneTZ) [reply apps/back-end/controllers/record-controller.js:25]: I'll add the authentication process in the next PR.

File: apps/back-end/controllers/record-controller.js (hunk 3 above)
Comment (AdrieneTZ) [reply apps/back-end/controllers/record-controller.js:96]: I'll add the authentication process in the next PR.

**Pull request merged**